core/interfaces/i_procedure_model.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * The interface for the data model of a procedure.
+ *
+ * @namespace Blockly.IProcedureModel
+ */
+
+import {IParameterModel} from './i_parameter_model.js';
+
+
+/**
+ * A data model for a procedure.
+ */
+export interface IProcedureModel {
+  /** Sets the human-readable name of the procedure. */
+  setName(name: string): IProcedureModel;
+
+  /**
+   * Inserts a parameter into the list of parameters.
+   *
+   * To move a parameter, first delete it, and then re-insert.
+   */
+  insertParameter(parameterModel: IParameterModel, index: number):
+      IParameterModel;
```

Comment:
Did you decide about the return type here?  Given that one could put a bunch of fluent method calls on the ParameterModel in the first argument, I don't see a compelling reason to return the `IParameterModel`.  It seems to me more useful to be able to add multiple arguments fluently (though I suppose that in practice this would probably be done one-per-iteration in a loop).

Reply:
Discussed and decided on `IProcedureModel`